testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -14,29 +14,190 @@
+                        "DATE '2005-09-10', " +  // a_date
+                        "TIMESTAMP '2005-09-10 13:00:00', " + // a_timestamp_seconds
+                        "TIMESTAMP '2005-09-10 13:00:00.123', " + // a_timestamp_millis
+                        "TIMESTAMP '2005-09-10 13:00:00.123456', " + // a_timestamp_micros
+                        "TIMESTAMP '2005-09-10 13:00:00.123456789', " + // a_timestamp_nanos
+                        "'dummy')");
+        // negative values
+        onSpark().executeQuery(
+                "INSERT INTO " + sparkTableName + " VALUES (" +
+                        "false, " + // a_boolean
+                        "-128, " + // a_tinyint
+                        "-32768, " + // a_smallint
+                        "-1000000012, " + // an_integer
+                        "-1000000000000012, " + // a_bigint
+                        "-10000000.123, " + // a_real
+                        "-100000000000.123, " + // a_double
+                        "CAST('-1234567.8901' AS decimal(11, 4)), " + // a_short_decimal
+                        "CAST('-1234567890123456789.0123456' AS decimal(26, 7)), " + // a_short_decimal
+                        "'', " + // a_string
+                        "DATE '1965-09-10', " + // a_date
+                        "TIMESTAMP '1965-09-10 13:00:00', " + // a_timestamp_seconds
+                        "TIMESTAMP '1965-09-10 13:00:00.123', " + // a_timestamp_millis
+                        "TIMESTAMP '1965-09-10 13:00:00.123456', " + // a_timestamp_micros
+                        "TIMESTAMP '1965-09-10 13:00:00.123456789', " + // a_timestamp_nanos
+                        "'dummy')");
+
+        List<Row> expected = List.of(
+                row(nCopies(16, null).toArray()),
+                row(
+                        true, // a_boolean≥
+                        (byte) 127, // a_tinyint
+                        (short) 32767, // a_smallint
+                        1000000000, // an_integer
+                        1000000000000000L, // a_bigint
+                        10000000.123F, // a_real
+                        100000000000.123, // a_double
+                        new BigDecimal("1234567.8901"), // a_short_decimal
+                        new BigDecimal("1234567890123456789.0123456"), // a_long_decimal
+                        "some string", // a_string
+                        java.sql.Date.valueOf(LocalDate.of(2005, 9, 10)), // a_date
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0)), // a_timestamp_seconds
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0, 123_000_000)), // a_timestamp_millis
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_micros
+                        java.sql.Timestamp.valueOf(LocalDateTime.of(2005, 9, 10, 13, 0, 0, 123_456_000)), // a_timestamp_nanos TODO/note nanos were truncated
```

Comment:
reminder to extend todo comment

Reply:
will reword. Spark timestamp has microseconds precision only.